### ISSUE TITLE
RTSP Stream Grabbing?

### ISSUE THREAD

**Issue by ciordia9** [node-alarm-dot-com/homebridge-node-alarm-dot-com#127]
Is there a way to grab or passthrough the rtsp streams? I see the ports are exposed on all these units from door to wall cams and the web apps have to be grabbing them somehow. It would be nice to pull those streams in and have them available.

**Comment by chase9** [comment]
I don't think this would be trivially possible and part of this project. From what I can tell, the firmware on these devices opens an OpenVPN connection to ADC and uses that to maintain a connection. They also randomly generate a root password and do not make it available to the end user. All of this is good from a security standpoint but makes it hard for owners to MiTM the cams :P

Breaking into your own equipment is outside of the scope of this project (we need to stay on the up and up with ADC!), but it would be in scope to set up a relay using the ADC proxy.

**Comment by chase9** [comment]
Notes: the web viewer makes a call to `https://www.alarm.com/web/api/video/videoSources/liveVideoHighestResSources/<CAM-ID>` and receives a response which includes `proxyUrl` under it's `attributes`. This proxy url can be used to view the camera live through ADC's servers.
